pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -171,25 +178,84 @@ export class ExtensionImpl implements Extension {
 
     await Promise.all(paths.map(async(p) => {
       try {
-        await this.client.copyFile(this.id, p, binDir);
+        await this.client.copyFile(this.id, p, binDir, { namespace: this.extensionNamespace });
       } catch (ex: any) {
         throw new ExtensionErrorImpl(ExtensionErrorCode.FILE_NOT_FOUND, `Could not copy host binary ${ p }`, ex);
       }
     }));
   }
 
+  get containerName() {
+    const normalizedId = this.id.toLowerCase().replaceAll(/[^a-z0-9_-]/g, '_');
+
+    return `rd-extension-${ normalizedId }`;
+  }
+
+  /**
+   * Extra the Docker Compose files into a newly-created temporary directory.
+   * This is required because nerdctl doesn't seem to keep the definition around,
+   * @note The caller is expected to remove the output directory.
+   * @param composePath the value of metadata.vm.composefile
+   */
+  protected async extractComposeDefinition(composePath: string): Promise<string> {
+    const workDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-ext-install-'));
+    const composeDir = path.posix.dirname(path.posix.normalize(composePath));
+
+    await this.client.copyFile(
+      this.id,
+      composeDir === '.' ? '/' : `${ composeDir }/`,
+      workDir,
+      { namespace: this.extensionNamespace });
+
+    return workDir;
+  }
+
+  protected async installContainers(workDir: string, metadata: ExtensionMetadata): Promise<void> {
```

Comment:
My editor is saying that the `workDir` argument is never used. Can we eliminate it?

Reply:
We can; I had it there so that it's the same as all the other `install*` methods, but that was mostly because I was using `Promise.allSettled` previously.